src/components/nabla/Pools/Backstop/AddLiquidity/useAddLiquidity.ts

```diff
@@ -1,56 +1,102 @@
+import { Big } from 'big.js';
+
 import { cacheKeys } from '../../../../../constants/cache';
 import { backstopPoolAbi } from '../../../../../contracts/nabla/BackstopPool';
-import { useGetAppDataByTenant } from '../../../../../hooks/useGetAppDataByTenant';
 import { useModalToggle } from '../../../../../services/modal';
-import { decimalToNative, FixedU128Decimals } from '../../../../../shared/parseNumbers/metric';
-import { useContractBalance } from '../../../../../shared/useContractBalance';
-import { useContractWrite } from '../../../../../shared/useContractWrite';
-import schema from './schema';
-import { AddLiquidityValues } from './types';
+import { decimalToRaw } from '../../../../../shared/parseNumbers/metric';
+import { erc20WrapperAbi } from '../../../../../contracts/nabla/ERC20Wrapper';
+import { useErc20ContractBalance } from '../../../../../hooks/nabla/useErc20ContractBalance';
+import { useContractWrite } from '../../../../../hooks/nabla/useContractWrite';
+
+interface AddLiquidityValues {
+  amount: string;
+}
+
+const schema = Yup.object<AddLiquidityValues>().shape({
+  amount: Yup.string().required(),
+});
 
-export const useAddLiquidity = (poolAddress: string, tokenAddress: string) => {
+export const useAddLiquidity = (
+  backstopPoolAddress: string,
+  tokenAddress: string,
+  poolTokenDecimals: number,
+  lpTokenDecimals: number,
+) => {
   const queryClient = useQueryClient();
-  const { indexerUrl } = useGetAppDataByTenant('nabla').data || {};
   const toggle = useModalToggle();
 
-  const balanceQuery = useContractBalance({ contractAddress: tokenAddress, decimals: FixedU128Decimals });
-  const depositQuery = useContractBalance({ contractAddress: poolAddress, decimals: FixedU128Decimals });
+  const balanceQuery = useErc20ContractBalance(erc20WrapperAbi, {
+    contractAddress: tokenAddress,
+    decimals: poolTokenDecimals,
+  });
+
+  const depositQuery = useErc20ContractBalance(backstopPoolAbi, {
+    contractAddress: backstopPoolAddress,
+    decimals: lpTokenDecimals,
+  });
 
   const form = useForm<AddLiquidityValues>({
     resolver: yupResolver(schema),
-    defaultValues: {},
+    defaultValues: {
+      amount: undefined,
+    },
   });
 
   const mutation = useContractWrite({
     abi: backstopPoolAbi,
-    address: poolAddress,
+    address: backstopPoolAddress,
     method: 'deposit',
-    onError: () => {
-      // ? log error - alert not needed as the transaction modal displays the error
-    },
-    onSuccess: () => {
-      form.reset();
-      balanceQuery.refetch();
-      depositQuery.refetch();
-      queryClient.refetchQueries([cacheKeys.backstopPools, indexerUrl]);
+    mutateOptions: {
+      onError: () => {
+        // ? log error - alert not needed as the transaction modal displays the error
+      },
+      onSuccess: () => {
+        form.reset();
+        balanceQuery.refetch();
+        depositQuery.refetch();
+        setTimeout(() => {
```

Comment:
Using `setTimeout` to refetch queries is not the most efficient way to handle data refetching. 
Instead we could maybe make the onSuccess `async` and put `await` before balanceQuery and depositQuery to make sure they're finished before the `queryClient.refetchQueries` is executed.
We could also try to put the `queryClient.refetchQueries([cacheKeys.backstopPools, indexerUrl])` in `onSettled` as it runs after `onSuccess` ( but it is also triggered after `onError` )

Reply:
This was not put there by me originally. There was a comment somewhere but I removed it. The reason to delay this is not to wait for `balanceQuery` and `depositQuery` to finish. The call to the refetch the `nablaInstance` is totally independent. The first two query directly from the smart contract on the blockchain. The `nablaInstance` comes from the indexer. We observed during tests that the indexer is always a little delayed. For that reason we wait a little to have a higher chance that it already delivers the new data.

Reply:
Perhaps we could extract this setTimeout for 2000ms to a separate function and reuse it wherever needed. This way, if we need to make changes in the future, we can do so in one place rather than having to search throughout the application. What do you think about this approach?

Reply:
I extracted it to a function `refetchDelayed`.

Reply:
Thank you @TorstenStueber